test/miscellaneous/Makefile.am

```diff
@@ -10,3 +10,35 @@ odp_api_from_cpp_SOURCES = odp_api_from_cpp.cpp
 noinst_PROGRAMS = odp_api_headers
 odp_api_headers_CFLAGS = $(AM_CFLAGS) -Wconversion
 odp_api_headers_SOURCES = odp_api_headers.c
+
+PKGCONFIG = PKG_CONFIG_PATH=$(libdir)/pkgconfig:$$PKG_CONFIG_PATH pkg-config --cflags --libs
+
+if enable_shared
+
+PROGRAM_shared = odp_api_headers_shared
+
+installcheck-local: $(PROGRAM_shared)
+
+$(PROGRAM_shared): $(srcdir)/$(odp_api_headers_SOURCES)
+	$(CC) $(AM_CFLAGS) $(CFLAGS) $^ -o $@ \
+		`$(PKGCONFIG) libodphelper` `$(PKGCONFIG) libodp-linux`
```

Comment:
Could the ODP library name (`libodp-linux`) be passed using a variable? Now other platform will need to patch this makefile manually.

Reply:
Yes, that's how it was supposed to be. Good catch, I somehow missed that. Fixed in v4.